WIO_EMW3166 target doesn't compile with blinky-baremetal

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

WIO_EMW3166  target fails to compile blinky-baremetal in Mbed OS 6

GCC and Arm Error

```

[Fatal Error] TLSSocketWrapper.h@28,10: mbedtls/platform.h: No such file or directory
[ERROR] In file included from ./mbed-os/features/netsocket/nsapi.h:42,
                 from ./mbed-os/features/netsocket/EthernetInterface.h:20,
                 from ./mbed-os/targets/TARGET_WICED/wiced_interface/WicedInterface.h:21,
                 from ./mbed-os/targets/TARGET_WICED/wiced_interface/default_wifi_interface.cpp:18:
./mbed-os/features/netsocket/TLSSocketWrapper.h:28:10: fatal error: mbedtls/platform.h: No such file or directory
   28 | #include "mbedtls/platform.h"
```

#### Target(s) affected by this defect ?

WIO_EMW3166

#### Toolchain(s) (name and version) displaying this defect ?

Arm Compiler 6 and GCC 9

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-6.0.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Mbed CLI 1.10.4

#### How is this defect reproduced ? 

```
mbed import https://github.com/ARMmbed/mbed-os-example-blinky-baremetal
cd mbed-os-example-blinky
mbed compile -t <toolchain> -m <target>
```

@ARMmbed/team-seeed @toyowata

--- a/targets/TARGET_WICED/wiced_interface/WicedInterface.h
+++ b/targets/TARGET_WICED/wiced_interface/WicedInterface.h
@@ -17,6 +17,8 @@
 #ifndef WICED_INTERFACE_H
 #define WICED_INTERFACE_H
 
+#if defined(MBED_CONF_NSAPI_PRESENT)
+
 #include "mbed.h"
 #include "EthernetInterface.h"
 #include "netsocket/OnboardNetworkStack.h"
@@ -109,5 +111,7 @@ private:
     char _pass[64]; /* The longest allowed passphrase + 1 */
     nsapi_security_t _security;
 };
+
+#endif
 
 #endif

--- a/targets/TARGET_WICED/wiced_interface/default_wifi_interface.cpp
+++ b/targets/TARGET_WICED/wiced_interface/default_wifi_interface.cpp
@@ -15,6 +15,8 @@
  * limitations under the License.
  */
 
+#if defined(MBED_CONF_NSAPI_PRESENT)
+
 #include "WicedInterface.h"
 
 WiFiInterface *WiFiInterface::get_target_default_instance()
@@ -22,3 +24,5 @@ WiFiInterface *WiFiInterface::get_target_default_instance()
     static WicedInterface wifi;
     return &wifi;
 }
+
+#endif

